CRM/Core/BAO/MailSettings.php

```diff
@@ -128,15 +128,16 @@ public static function add(&$params) {
     }
 
     if (empty($params['id'])) {
-      $params['is_ssl'] = CRM_Utils_Array::value('is_ssl', $params, FALSE);
-      $params['is_default'] = CRM_Utils_Array::value('is_default', $params, FALSE);
+      $params['is_ssl'] = $params['is_ssl'] ?? FALSE;
+      $params['is_default'] = $params['is_default'] ?? FALSE;
+      $params['domain_id'] = $params['domain_id'] ?? CRM_Core_Config::domainID();
     }
 
     //handle is_default.
     if (!empty($params['is_default'])) {
+      $domain = $params['domain_id'] ?? CRM_Core_DAO::getFieldValue(__CLASS__, $params['id'], 'domain_id');
```

Comment:
We can get away with assuming `$params['id']` is set here because the latter half of this `??` expression is only reachable if `domain_id` is not set, and that can only happen if `$params['id']` is not empty per above.